airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -69,8 +69,15 @@ async def infer_schema(
                 self._handle_unprocessable_file(file, format, logger)
 
             return {
-                "content": {"type": "string"},
-                "document_key": {"type": "string"},
+                "content": {
+                    "type": "string",
+                    "description": "Content of the file as markdown. Might be null if the file could not be parsed",
+                },
+                "document_key": {"type": "string", "description": "Unique identifier of the document, e.g. the file path"},
+                "error": {
+                    "type": "string",
+                    "description": "Error message if the file could not be parsed even though the file is supported",
+                },
```

Comment:
It feels like meta / high-level failures like this at the parser level might warrant an `_ab_` column like `_ab_read_error`. It would be nice if there were a precedent on column naming here. 

Do we know of any similar cases where a 'failure to parse' error is passed from source to destination? And, separately, what do you think of using an `_ab_` name prefix for this?

Reply:
The normalization logic is creating `_airbyte_meta` which contains an error array (https://airbyte.com/blog/introducing-airbyte-destinations-v2-typing-deduping).

However, I think we should stay consistent with the existing `_ab_source_file_url` and `_ab_source_file_last_modified`. This means it should be called `ab_source_file_parse_error`.

What do you think @clnoll ? I know it's different than the other parsers, but here we have a 1:1 mapping between file and record so this approach seems to make sense.

Reply:
Couple of thoughts - 

What's the argument for making this logic specific to the unstructured parser? Maybe I'm not familiar enough with the use case, but I'm hesitant to have the parsers' output deviate.

My initial feeling is that if we're only adding this to the unstructured parser, I'd prefer that it not be a top-level `_ab_` column, as those tend to have specific use cases across connectors, but as far as considerations like breaking changes go, I assume the issue would be the same whether it's here or in a top-level `_ab_` column. I'd be interested to know what @girarda thinks about this too though.

I assume that there are no current plans to have a 1:many mapping between file & record, but could definitely see it happening in the future (e.g. providing an option to breakdown by page, etc.). Probably a good idea to think ahead about how that might affect us here.

Reply:
To a user, the ab prefix indicates it's not the actual payload of the record, but some piece of meta data. I think it's a good fit in that sense. I don't know how it would fit for other parsers, maybe we can make it a thing just for the unstructured parser now and widen it to all parsers once we have a use case?

That keeps the surface for breaking changes lower as well

Reply:
@clnoll is the concern about the naming or the fact that it is a top level field? Different parsers yield different schemas (eg schema vs schemaless) so I'm not opposed to adding custom columns for unstructured.

> I don't know how it would fit for other parsers, maybe we can make it a thing just for the unstructured parser now and widen it to all parsers once we have a use case?

Agreed.